### DISCUSSION TITLE
Does some part of VanJS make use of CommonJS module syntax?

### DISCUSSION THREAD

**Discussion by Ashvith10** [vanjs-org/van#399] [General]
Like for example, this code snippet, I can't help but notice how it resembles CommonJS syntax:

```js
const {a, div, p} = van.tags
```

**Comment by Tao-VanJS** [comment]
Not really. These are just standard JavaScript syntax for [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

**Reply by Ashvith10** [reply]
@Tao-VanJS sorry for not replying sooner, and thanks for answering. Usually, as we see in other framework libraries, we import components, right, but in this circumstance, we don't - why so?

**Reply by Tao-VanJS** [reply]
I don't quite understand the question. In

```js
const {a, div, p} = van.tags
```

`a`, `div`, `p` are not components that need to be imported. They are just tag functions that can create native DOM elements.

**Reply by Ashvith10** [reply]
@Tao-VanJS I may be in the wrong here, but I was wondering - since we are importing "van", which includes a whole bunch of stuff including "tags", which in turn has a bunch of tag replacement for the HTML equivalent, wouldn't it be computationally expensive?

**Reply by Tao-VanJS** [reply]
No, `van.tags` is just a `Proxy` object which returns the same function with different bindings based on the property name: https://github.com/vanjs-org/van/blob/e21df7d6af722b85e230b24c47cb0e406dd146ec/src/van.js#L137
